api/src/endpoints/neo4j.js

```diff
@@ -143,23 +162,36 @@ neo4jRoutes.get('/3d-network', async (req, res) => {
       payload = { ...payload, type, id };
     }
 
+    validatePayload(payload);
+
     const network = await get3dNetwork(payload);
     res.json(network);
   } catch (e) {
     res.status(400).send(e.message);
   }
 });
 
+function validateJson(json) {
+  if (typeof json === 'object') {
+    Object.values(json).forEach(val => validateJson(val));
```

Comment:
I assume that here you wanted to call validateInput() ?

Reply:
No, I think I meant it to be recursive.

Reply:
Of course, if you pass a circular structure to this function it will not terminate but JSON structures can't be circular. I do not know if this should be considered a problem.